compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -1217,8 +1217,19 @@ trait Checking {
   /** 1. Check that all case classes that extend `scala.reflect.Enum` are `enum` cases
    *  2. Check that parameterised `enum` cases do not extend java.lang.Enum.
    *  3. Check that only a static `enum` base class can extend java.lang.Enum.
+   *  4. Check that user does not implement an `ordinal` method in the body of an enum class.
```

Comment:
we require that `ordinal` is not user implemented because `Mirror.SumOf`  synthesis delegates to `ordinal` for enums. It's worth noting that in 3.1.0 it was already an error as the child cases would override a user defined ordinal without the `override` keyword